### DISCUSSION TITLE
PCAP Retention Settings

### DISCUSSION THREAD

**Discussion by out-n-about** [Security-Onion-Solutions/securityonion#2741] [Unsupported Versions] [answered]
I have an instance set up in AWS.  I have 1TB set up for the storage but according to grafana its only using about 1/3 of it and saying that the retention is 3 weeks.  It has been on longer than that but it always seems to say the retention is 3 weeks.  Is there a setting somewhere that is keeping it at the value?

**Comment by weslambert** [comment]
There may be an error with the stats updating in Grafana.  Let me check on that

**Reply by out-n-about** [reply]
I ran df -h to see the current drive usage and it appears that grafana is showing correct.  only 33% of the drive is in use.  Thats as high as I've ever seen it.  Total drive size is 1 TB.

**Comment by weslambert** [comment]
After looking at this with the rest of the team, it seems there could be a few things going on. You'll want to reduce log_size_limit `(/opt/so/saltstack/local/pillar/minions/$node.sls)` to 50% of available disk space (in GB) for /nsm. Additionally, you may need to check the size of the Strelka log in /nsm/strelka/log , removing it and restarting Strelka if necessary (can get quite large -- we are working on rotation). Last, you may have quite a few so-sensor-clean processes running. So, you may need to modify the so-sensor-clean script as detailed here: 9448177 in order to prevent itself from running multiple instances, then clean up the existing ones, if necessary (find them with ps aux | grep sensor-clean | grep -v grep, then kill the processes, or reboot). These items should be fixed by default in the next release.

**Comment by c9845** [comment]
Please see https://github.com/Security-Onion-Solutions/securityonion/discussions/3591 as a possible solution (editing the `MaxDirectoryFiles` setting).

**Comment by dougburks** [comment, marked as answer]
Added to documentation:
https://docs.securityonion.net/en/2.3/stenographer.html#maxdirectoryfiles
